src/core/Accordion.tsx

```diff
@@ -117,8 +118,11 @@ const AccordionRow = ({
       >
         {rowIcon ? (
           <Icon
-            name={rowIcon}
+            name={typeof rowIcon === "object" ? rowIcon.name : rowIcon}
```

Comment:
[silly question] why not do a proper typescript check? (Not that I know the syntax off by heart)

Reply:
It's a bit long winded to do type guards in TS, it would look something like this which pretty much does what I'm doing here. Could make sense for more complex types but not this example I think
```ts
const isAccordionIcon = (icon: IconName | AccordionIcon): icon is AccordionIcon => {
  return typeof icon === 'object' && 'name' in icon;
};
```